samza-sql/src/main/java/org/apache/samza/sql/translator/JoinTranslator.java

```diff
@@ -250,7 +251,17 @@ private String dumpRelPlanForNode(RelNode relNode) {
         SqlExplainLevel.EXPPLAN_ATTRIBUTES);
   }
 
-  private SqlIOConfig resolveSourceConfig(RelNode relNode) {
+  private SqlIOConfig resolveSourceConfigForTable(RelNode relNode) {
+    if (relNode instanceof LogicalProject) {
+      return resolveSourceConfigForTable(((LogicalProject) relNode).getInput());
+    }
+
+    // For stream-table-table join, we will run into a join relNode. The output of stream-table join is a stream.
+    // In that case, let's return null, as the source for relNode is not a table but a stream.
+    if (relNode.getInputs().size() > 1) {
+      return null;
+    }
+
     String sourceName = String.join(".", relNode.getTable().getQualifiedName());
     SqlIOConfig sourceConfig = ioResolver.fetchSourceInfo(sourceName);
     if (sourceConfig == null) {
```

Comment:
when is SourceConfig null?

Reply:
This is an unexpected error condition. We expect the source configs to be set.